### ISSUE TITLE
[Environment Variables] Base environment tag not saved

### ISSUE THREAD

**Issue by viico** [getinsomnia/insomnia#5858]
### Expected Behavior

Tag body attribute saved when we change `Request` field.

### Actual Behavior

When I change the `Request` in a tag (type `Body Attribute`) present in my base environment it is not saved. If i close the modal and reopen the request is still the one before à change it.

### Reproduction Steps

1. Set a variable with a tag (type `Body Attribute`) in the base environment
2. Close the environment modal
3. Reopen the base environment and change the request in the tag
4. Close the environment modal
5. Reopen the base environment => it's not the new request selectec

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Insomnia Version

2023.1.0

### What operating system are you using?

Windows

### Operating System Version

Windows 10 Entreprise

### Installation method

download from insomnia.rest

**Comment by viico** [comment]
One workaround is to modify the base environment (add / modify / remove another variable).

**Comment by mhmmd-ysf** [comment]
I'm having the same issue here

Version: Insomnia 2023.1.0
Build date: 3/9/2023
OS: Windows_NT x64 10.0.22000
Electron: 22.0.0
Node: 16.17.1
V8: 10.8.168.20-electron.0
Architecture: x64

**Comment by fanch317** [comment]
I have same issue here.

This issue only happens when you update a parameter on a function (tested with `Body Attribute` and a `SHA512`)

![Recording 2023-06-29 at 12 02 00](https://github.com/Kong/insomnia/assets/1344319/b9dd20c8-8d18-42ff-ba8e-46a6ea565ade)


```
Version: Insomnia 2023.3.0
Build date: 27/06/2023
OS: Windows_NT x64 10.0.19045
Electron: 22.3.11
Node: 16.17.1
V8: 10.8.168.25-electron.0
Architecture: x64
```
Related to #5612

**Comment by tlund101** [comment]
This is fixed by https://github.com/Kong/insomnia/pull/6011 A workaround is to just click in the json and click back out of it to trigger the `onBlur` event.

**Comment by mikebridge** [comment]
This behaviour is recurring for me after upgrading to `2023.5.7`.  the `onBlur` workaround doesn't seem to help---none of my changes are reflected after closing the editing window.

**Comment by mikebridge** [comment]
The only way I have found to get it to change variables is to close Insomnia entirely after editing a variable and reload the whole app.

**Comment by viico** [comment]
> The only way I have found to get it to change variables is to close Insomnia entirely after editing a variable and reload the whole app.

Hi, my precedent workaround didn't work for you ?

> One workaround is to modify the base environment (add / modify / remove another variable).

**Comment by mhmmd-ysf** [comment]
> One workaround is to modify the base environment (add / modify / remove another variable).

Creating a dedicated property for env variable update seems to be the best option currently.

```json
{
  ...
  "dedicated_variable": "update this value if you only want to change env tags"
}
```

**Comment by mikebridge** [comment]
This appears to have reverted after upgrading to 8.5.0.  To reproduce, I do an `Edit Tag` on an existing `Response: reference values form other request's responses`, then select a `Request`, then change the filter on `Attribute: Body Attribute` to the value `$.access_token` and verify `Live Preview` has a value, then click `Done`.

When I reopen, all my changes are gone.

**Comment by gatzjames** [comment]
Closing as this should now be fixed in 8.5.1 by #6941
Thanks everyone for the feedback and especially @mikebridge for bringing it up on the updates of the environment modal.
